src/main/java/com/lothrazar/storagenetwork/item/ItemCollector.java

```diff
@@ -60,12 +59,16 @@ public void onEntityItemPickupEvent(EntityItemPickupEvent event) {
         BlockEntity tile = serverTargetWorld.getBlockEntity(dp.getBlockPos());
         if (tile instanceof TileMain) {
           TileMain network = (TileMain) tile;
-          //
-          int countUnmoved = network.insertStack(item, false);
+
+          // Create a new reference to the stack, try to insert that into the
+          // network, then change the original stack size so the player picks up
+          // only what remains, if anything.
+          int countUnmoved = network.insertStack(item.copy(), false);
+          item.setCount(countUnmoved);
+
+          // We still want to play the pickup sound, even if Minecraft silently
+          // deletes the stack we just emptied.
           if (countUnmoved == 0) {
-            item.setCount(0);
-            event.getItem().setItem(item);
-            event.getItem().remove(Entity.RemovalReason.KILLED);
```

Comment:
This is why items were being voided in situations such as in #394. Deleting the stack from the world also deletes the item that was just inserted into the inventory network because they are both the same object.